tests/common/eventHandlers.ts

```diff
@@ -0,0 +1,114 @@
+import { ZkVerifyEvents, TransactionStatus } from '../../src';
+
+export interface EventResults {
+    includedInBlockEmitted: boolean;
+    finalizedEmitted: boolean;
+    errorEventEmitted: boolean;
+    attestationConfirmedEmitted?: boolean;
+    attestationBeforeExpectedEmitted?: boolean;
+    attestationMissedEmitted?: boolean;
+    broadcastEmitted?: boolean;
+    unsubscribeEmitted?: boolean;
+}
+
+export const handleCommonEvents = (
+    events: any,
+    expectedProofType: string,
+    expectedDataType: 'verify' | 'vkRegistration',
+): EventResults => {
+    const eventResults: EventResults = {
+        includedInBlockEmitted: false,
+        finalizedEmitted: false,
+        errorEventEmitted: false,
+        attestationConfirmedEmitted: false,
+        attestationBeforeExpectedEmitted: false,
+        attestationMissedEmitted: false,
+        broadcastEmitted: false,
+        unsubscribeEmitted: false,
+    };
+
+    events.on(ZkVerifyEvents.ErrorEvent, () => {
+        eventResults.errorEventEmitted = true;
+    });
+
+    events.on(ZkVerifyEvents.IncludedInBlock, (eventData: any) => {
+        eventResults.includedInBlockEmitted = true;
+        expect(eventData).toBeDefined();
+        expect(eventData.blockHash).not.toBeNull();
+        expect(eventData.proofType).toBe(expectedProofType);
+        expect(eventData.status).toBe(TransactionStatus.InBlock);
+        expect(eventData.txHash).toBeDefined();
+
+        if (expectedDataType === 'vkRegistration') {
+            expect(eventData.statementHash).toBeDefined();
+            expect(eventData.attestationId).toBeUndefined();
+            expect(eventData.leafDigest).toBeUndefined();
+        } else if (expectedDataType === 'verify') {
+            expect(eventData.attestationId).not.toBeNull();
+            expect(eventData.leafDigest).not.toBeNull();
+            expect(eventData.statementHash).toBeUndefined();
```

Comment:
The registered vk event emits simply H(vk) while the statementHash is something emitted by the ProofVerified event and it's H(prefix, H(vk), H(pubs)). It seems to me here that:
- If "vkRegistration", then statementHash must be undefined while an additional param, vkHash must be present
- If "verify", then statementHash must be defined. Keep in mind that statementHash and leafDigest are exactly the same thing in this context.

Reply:
These tests are checking the format the zkverifyjs package is responding with and confirming the packages logic is correctly mapping data for the events received rather than testing the pure format returned by zkverify itself.

These tests pass - is there something incorrect in the logic?

Reply:
In the second if, leafDigest and statementHash are the same field, so no reason for them to be different.
In the first if, statementHash shall be replaced with vkHash.
I understand that these are tests, but nevertheless should provide guidance for developers (us or third party) looking for guidance on how to use the package, so the most accurate we are, the better imho

Reply:
I still dont understand, is the confusion around the interface types being used to represent this data?  Is the naming of the fields not appropriate?

`export interface TransactionInfo {
  blockHash: string;
  proofType: ProofType;
  status: TransactionStatus;
  txHash?: string;
  extrinsicIndex?: number;
  feeInfo?: {
    payer: string;
    actualFee: string;
    tip: string;
    paysFee: string;
  };
  weightInfo?: {
    refTime?: string;
    proofSize?: string;
  };
  txClass?: string;
}

export interface VerifyTransactionInfo extends TransactionInfo {
  attestationId: number | undefined;
  leafDigest: string | null;
  attestationConfirmed: boolean;
  attestationEvent?: AttestationEvent;
}

export interface VKRegistrationTransactionInfo extends TransactionInfo {
  statementHash?: string;
}`

Reply:
The test is checking the return types, statementHash should not be defined for verify because the type is of VerifyTransactionInfo. Its just a check to make sure that is the case.

Same for attestationId and leafDigest on the VKRegistrationTransactionInfo type, they should not exist.